substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -633,4 +656,648 @@ mod e2e {
+		ExtBuilder::signed().build_and_execute(|| {
+			roll_to_signed_open();
+			assert_full_snapshot();
+
+			// Submit only an invalid solution (register but don't submit pages)
+			let invalid_score =
+				ElectionScore { minimal_stake: 10, sum_stake: 10, sum_stake_squared: 100 };
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(99), invalid_score));
+
+			let current_round = SignedPallet::current_round();
+			assert_eq!(Submissions::<Runtime>::submitters_count(current_round), 1);
+
+			roll_to_signed_validation_open();
+			let _ = verifier_events(); // Clear events
+
+			roll_to_full_verification();
+
+			// Check that rejection events were properly generated
+			let rejection_events = verifier_events_since_last_call();
+			assert_eq!(
+				rejection_events,
+				vec![
+					crate::verifier::Event::Verified(2, 0),
+					crate::verifier::Event::Verified(1, 0),
+					crate::verifier::Event::Verified(0, 0),
+					crate::verifier::Event::VerificationFailed(0, FeasibilityError::InvalidScore),
+				]
+			);
+
+			// Verify no-restart conditions are met
+			assert!(crate::Pallet::<Runtime>::current_phase().is_signed_validation());
+			assert!(!Submissions::<Runtime>::has_leader(current_round));
+			assert_eq!(Submissions::<Runtime>::submitters_count(current_round), 0);
+			assert_eq!(VerifierPallet::status(), crate::verifier::Status::Nothing);
+
+			// Verifier should remain idle for the rest of the signed validation phase
+			while crate::Pallet::<Runtime>::current_phase().is_signed_validation() {
+				roll_next();
+				assert_eq!(VerifierPallet::status(), crate::verifier::Status::Nothing);
+			}
+
+			// Check that expected events were emitted for the rejection
+			assert_eq!(
+				signed_events(),
+				vec![Event::Registered(0, 99, invalid_score), Event::Slashed(0, 99, 5),]
+			);
+
+			roll_next();
+			assert_eq!(VerifierPallet::status(), crate::verifier::Status::Nothing);
+		});
+	}
+
+	#[test]
+	fn after_accepting_one_solution_verifier_is_idle_if_no_leader_exists() {
+		ExtBuilder::signed().build_and_execute(|| {
+			roll_to_signed_open();
+			assert_full_snapshot();
+
+			// Load and verify a good solution
+			let good_solution = mine_full_solution().unwrap();
+			load_signed_for_verification_and_start_and_roll_to_verified(
+				99,
+				good_solution.clone(),
+				0,
+			);
+
+			assert_eq!(
+				signed_events(),
+				vec![
+					Event::Registered(0, 99, good_solution.score),
+					Event::Stored(0, 99, 0),
+					Event::Stored(0, 99, 1),
+					Event::Stored(0, 99, 2),
+					Event::Rewarded(0, 99, 7),
+				]
+			);
+
+			// Check only the verification events for this solution
+			let solution_events = verifier_events_since_last_call();
+			assert_eq!(
+				solution_events,
+				vec![
+					crate::verifier::Event::Verified(2, 2),
+					crate::verifier::Event::Verified(1, 2),
+					crate::verifier::Event::Verified(0, 2),
+					crate::verifier::Event::Queued(good_solution.score, None),
+				]
+			);
+
+			roll_to_done();
```

Comment:
This is not exactly consistent with the previous test, the previous one was more detailed. 

So at this point of the code, you are still in SignedValidation. Similar to the previous test, you can `roll_next` in a
while loop until `SignedValidation` is over, and make sure we remain in `Status::Nothing`. You can also add a
`assert!(!Submissions::<Runtime>::has_leader(current_round));` again to hint _this is why we remain in
`Status::Nothing`_.

Saying this because for all I know, when just say `roll_to_done`, internally the verifier pallet might do one more
verification internally.